[WIP] AAP-4102 HA Hub on SELinux

TarikFD opened the pull request:
Jira: https://issues.redhat.com/browse/AAP-4102

Documenting how to mount an NFS server on SELinux to deploy a HA automation hub. 

Preview (VPN required): http://file.rdu.redhat.com/kevchin/ha-hub-selinux/tmp/en-US/html-single/#assembly-high-availability-hub-selinux

mikedep333 commented on downstream/assemblies/hub/assembly-high-availability-hub-selinux.adoc:
```diff
@@ -0,0 +1,88 @@
+----
+$ mount -a
+----
+. Edit `pulpcore.service` and add the following entry to ???:
```
We should tell them to run the command:
`systemctl edit pulpcore.service`

mikedep333 commented on downstream/assemblies/hub/assembly-high-availability-hub-selinux.adoc:
```diff
@@ -0,0 +1,88 @@
+----
+$ umount /var/lib/pulp
+----
+. Open `/etc/fstab` using a text editor, then add the following contexts to each mount point:
++
+----
+srv_rhel8:/data /var/lib/pulp nfs defaults,_netdev,context="system_u:object_r:pulpcore_var_lib_t:s0" 0 0
+srv_rhel8:/data/pulpcore_static /var/lib/pulp/pulpcore_static nfs defaults,_netdev,context="system_u:object_r:httpd_sys_content_rw_t:s0" 0 0
```
This is the only way I know of to resolve the problem currently, but it is an incomplete fix. Because there will be SELinux errors during installation. Users would have to set it to permissive during installation.

I am looking into possible solutions.

mikedep333 commented on downstream/assemblies/hub/assembly-high-availability-hub-selinux.adoc:
```diff
@@ -0,0 +1,82 @@
++
+----
+srv_rhel8:/data /var/lib/pulp nfs defaults,_netdev 0 0
+srv_rhel8:/data/pulpcore_static /var/lib/pulp/pulpcore_static nfs defaults,_netdev 0 0
```
Also
> To mount multiple mounts from a single NFS export, with each mount having a different context, use the -o nosharecache,context options".
https://www.linuxtopia.org/online_books/rhel6/rhel_6_selinux/rhel_6_selinux_sect-Security-Enhanced_Linux-Mounting_File_Systems-Multiple_NFS_Mounts.html

I tested that "nosharecache" resolves the issue.

So please instruct the users to add "nosharecache" all along.

TarikFD commented:
@mikedep333 please review the latest changes. Apologies if I misunderstood your feedback since I'm quite unfamiliar with these commands. To sum up my changes:

- Edited the procedure so that the user can set the context for `pulpcore_static` at the start and removed the steps to unmount it
- Added `nosharecache` to the mounting command (please see line 46 to ensure I added it correctly)
-- Would I need to add `nosharecache` to the mount command for `pulpcore_static` too?
- Added a troubleshooting at the end to document the temporary fix as discussed in [AAP-4342](https://issues.redhat.com/browse/AAP-4342)

mikedep333 commented on downstream/assemblies/hub/assembly-high-availability-hub-selinux.adoc:
```diff
@@ -0,0 +1,83 @@
+srv_rhel8:/data /var/lib/pulp nfs defaults,_netdev 0 0
+srv_rhel8:/data/pulpcore_static /var/lib/pulp/pulpcore_static nfs defaults,_netdev,context="system_u:object_r:httpd_sys_content_rw_t:s0" 0 0
+----
+. Run the mount command:
++
+----
+$ mount -a
+----
```
So they actually need to run:
```
mount /var/lib/pulp
mkdir /var/lib/pulp/pulpcore_static
mount /var/lib/pulp/pulpcore_static
```
Because the folder needs to exist, and it cannot be created until after the 1st is mounted.

TarikFD commented on downstream/assemblies/hub/assembly-high-availability-hub-selinux.adoc:
```diff
@@ -0,0 +1,83 @@
+
+= Setting up a high availability automation hub on SELinux
+
+To set up a high availability (HA) deployment of {HubName} on SELinux, you will need to create two mount points, one for `/var/lib/pulp` and one for `/var/lib/pulp/pulpcore_static`, before you run the installer.
```
This sounds like a prerequisite, so maybe I can add a prereq at the start: "You have already configured a NFS export on your server" ?

TarikFD commented on downstream/assemblies/hub/assembly-high-availability-hub-selinux.adoc:
```diff
@@ -0,0 +1,83 @@
++
+----
+$ systemctl stop pulpcore.service
+----
+. Unmount `/var/lib/pulp/`:
++
+----
+$ umount /var/lib/pulp
+----
+. Open `/etc/fstab` using a text editor, then add the following context to the `/var/lib/pulp/` mount point:
++
+----
+srv_rhel8:/data /var/lib/pulp nfs defaults,_netdev,nosharecache,context="system_u:object_r:pulpcore_var_lib_t:s0" 0 0
+----
+. Run the mount command:
++
+----
+$ mount -a
+----
+. Edit `pulpcore.service` using `systemctl`:
++
+----
```
Just to clarify, since the new installer will handle the unmount, modify fstab, and remounting, the user can proceed from line 27 (run .setup.sh) to line 53 (Edit `pulpcore.service` using `systemctl`) ?

TarikFD commented:
Another round of feedback for you @mikedep333 . The changes this time:

- Added prereq about needing NFS export
- Restructured procedure since pulp_installer 3.18+ users only need to add context to `/var/lib/pulp/pulpcore_static`
- Added section for pulp_installer 3.17 and older users to manually add the context on `/var/lib/pulp` (hopefully I got my version numbers correct!)
